bot.js

```diff
@@ -1,197 +1,91 @@
-    console.log(spentCities);
+  if (session.length == 0 || index == -1) {
+    session[session.length] = new game.session(session.length, msg.chat.id);
+    index = session.length - 1;
 
-    bot.sendMessage(chatId, firstSymbToUpperCase(randCity));
-    return;
+    let start = game.start(index);
+    start.message.map(messageItem =>
+      bot.sendMessage(session[index].chatId, messageItem)
+    );
   }
+});
 
-  if (
-    gameStart &&
-    validMessage(chatId, city) &&
-    chekCityInDB(chatId, city, lastLetter, cities)
-  ) {
-    spentCities.add(city);
-    let sCity = selectCityByLetter(lastValidLetter(city), cities);
-    //4. Если не осталось городов то писать «Я проиграл»
-    if (sCity == -1 || sCity == undefined) {
-      bot.sendMessage(
-        chatId,
-        "Игра окончена, я проиграл! \nГорода которые были названы:\n" +
-          [...spentCities].join(", ")
-      );
-      bot.sendMessage(chatId, "Давай сыграем еще!");
-      session(false);
-      return;
-    } else {
-      spentCities.add(sCity);
-      console.log(spentCities);
-      lastLetter = lastValidLetter(sCity);
-      //3. Ответить городом
-      bot.sendMessage(chatId, firstSymbToUpperCase(sCity));
-    }
-  }
-}
+bot.onText(/сдаюсь/i, msg => {
+  let index = session.findIndex(
+    sessionItem => sessionItem.chatId === msg.chat.id
+  );
 
-function session(start) {
-  if (start) {
-    gameStart = true;
-  }
-  if (!start) {
-    gameStart = false;
-    spentCities.clear();
+  if (index >= 0) {
+    bot.sendMessage(
+      session[index].chatId,
+      "Я выиграл!!! \nГорода которые были названы:\n" +
+        [...session[index].spentCities].join(", ")
+    );
+    bot.sendMessage(session[index].chatId, "Давай еще сыграем!");
+    console.log("bot.js 56 session: ");
+    console.log(session);
+    session.splice(index, 1);
   }
-}
-
-function lastValidLetter(str) {
-  let letter = str[str.length - 1];
-  if (letter == "ь" || letter == "ъ") letter = str[str.length - 2];
-  return letter;
-}
+});
 
-function firstSymbToUpperCase(str) {
-  return str[0].toUpperCase() + str.slice(1);
-}
+bot.on("message", msg => {
+  if (usedKeyWords.includes(msg.text.toLowerCase())) return;
 
-function selectCityByLetter(letter, cities) {
-  let findCities = cities.filter(
-    item => item[0] == letter && !spentCities.has(item)
+  let index = session.findIndex(
+    sessionItem => sessionItem.chatId === msg.chat.id
   );
-  //Условие ниже не срабатывает.. когда фильтр ничего не находит он должен вернуть пустой массив,
-  //но проверка на пустой массив почему-то не срабатывает
-  if (findCities == [] || findCities == undefined || findCities == NaN) {
-    console.log("149 Город не найден");
-    return -1;
-  } else return randomCity(findCities);
-}
-
-function chekCityInDB(chatId, city, lastLetter, cities) {
-  if (cities.includes(city)) {
-    let findCity = cities.find(item => item[0] == lastLetter && item == city);
+  if (index >= 0 && msg.chat.id == session[index].chatId) {
```

Comment:
why do you need this check of `msg.chat.id` if you are looking for session using the same check inside `.find`

Reply:
for reinsurance. I will take it away..